File: framework/languages/java/zlink-framework-core/src/main/java/systems/zlink/framework/actors/ZLinkActorContext.java
package systems.zlink.framework.actors;

import java.util.Optional;
import systems.zlink.contracts.core.RoutingId;

public interface ZLinkActorContext {
    Optional<String> spotId();

    ZLinkBoundSession boundSession();

    ZLinkActorJoinCall joinSpot(String spotId, Object request);

    ZLinkActorJoinCall joinEntrySpot(RoutingId spotNodeRid, Object request);
}

File: framework/languages/node/packages/framework/src/contracts/Actors/ZLinkActorContext.ts
import type { RoutingId, SpotId } from '../Common';
import type { ZLinkBoundSession } from '../Streams';
import type { ZLinkActorHandlerRegistry } from '../Spots';
import type { ZLinkActorJoinEntrySpotCall, ZLinkActorJoinSpotCall } from './ZLinkActorFactory';

export interface ZLinkActorContext {
  readonly meshName: string;
  readonly spotId?: SpotId;
  readonly handlers: ZLinkActorHandlerRegistry;
  readonly boundSession: ZLinkBoundSession;
  joinSpot(spotId: SpotId, request: unknown): ZLinkActorJoinSpotCall;
  joinEntrySpot(nodeRid: RoutingId, request: unknown): ZLinkActorJoinEntrySpotCall;
  leaveSpot(signal?: AbortSignal): Promise<void>;
}
